fdbclient/SpecialKeySpace.actor.cpp

```diff
@@ -1441,14 +1441,15 @@ ACTOR Future<Optional<std::string>> globalConfigCommitActor(GlobalConfigImpl* gl
 	while (iter != ranges.end()) {
 		std::pair<bool, Optional<Value>> entry = iter->value();
 		if (entry.first) {
-			if (entry.second.present()) {
+			if (entry.second.present() && iter->begin().startsWith(globalConfig->getKeyRange().begin)) {
 				Key bareKey = iter->begin().removePrefix(globalConfig->getKeyRange().begin);
 				vh.mutations.emplace_back_deep(vh.mutations.arena(),
 				                               MutationRef(MutationRef::SetValue, bareKey, entry.second.get()));
 
 				Key systemKey = bareKey.withPrefix(globalConfigKeysPrefix);
 				tr.set(systemKey, entry.second.get());
-			} else {
+			} else if (!entry.second.present() && iter->range().begin.startsWith(globalConfig->getKeyRange().begin) &&
+			           iter->range().end.startsWith(globalConfig->getKeyRange().begin)) {
```

Comment:
Does the special key-space contract guarantee that an overlapping entry will be entirely contained in the desired key range?

Reply:
I did not follow the PR. But here looks like the code is dealing with `clear` on the special keys.
The contract for special keys' `clearrange` is that it should be in one implemented special key range, otherwise, an error will be thrown.
Not sure if this help clarify some confusion here.